[#572] Update Infrastructure Mappings list to display OSP mappings

mturley opened the pull request:
Closes #572

michaelkro commented on app/javascript/react/screens/App/Overview/components/InfrastructureMappingsList/helpers.js:
```diff
@@ -47,15 +47,19 @@ export const mapInfrastructureMappings = (transformation_mapping_items, clusters
+  });
+  const datastoreMappingItems = transformation_mapping_items.filter(item => {
+    const destination = item.destination_type.toLowerCase();
+    return destination === 'storage' || destination === 'cloudvolume';
```
Now that https://github.com/ManageIQ/manageiq-v2v/pull/673 is in, might as well update this now

mturley replied:
I'm on it

mturley commented:
@michaelkro , @vconzola said he will mock up some label changes for this too, so i'm waiting on that before we merge this.

mturley commented:
Ahh! Good catch... @michaelkro it looks like it's displaying the RHV version of all three (networks, datastores, clusters). This will be a tricky fix, I need to fetch `cloud_tenant`, `cloud_volume_type` and `cloud_network`, and possibly refactor the mapping code a bit.

mturley commented:
@vconzola, when you return on Monday I just want to run this by you. In addition to the labels you mocked up in #572, because this data was being displayed and it was easier than adding a bunch of conditions specific to OSP just to hide things, I added `Provider` to the target labels for datastores and networks, and I added `Cluster` to the source labels for datastores and networks. Let me know if these are okay:

![screenshot 2018-10-05 16 32 02](https://user-images.githubusercontent.com/811963/46559105-42ce7180-c8bd-11e8-9a89-c760d4721216.png)

![screenshot 2018-10-05 16 31 58](https://user-images.githubusercontent.com/811963/46559117-47932580-c8bd-11e8-8c33-4e33311ec173.png)

![screenshot 2018-10-05 16 31 52](https://user-images.githubusercontent.com/811963/46559123-4c57d980-c8bd-11e8-86b5-4bf590859a62.png)

vconzola commented:
@mturley This looks good. We'll need to create a separate issue to add the source provider type to the list view too.

mturley commented:
@vconzola I think we already have one: https://github.com/ManageIQ/manageiq-v2v/issues/681

vconzola commented:
@mturley Thanks. I forgot I'd already opened one.

michaelkro commented on app/javascript/packs/polyfills.js:
```diff
@@ -4,3 +4,4 @@
 import 'core-js/es6/promise';
 import 'core-js/es6/set';
 import 'core-js/es6/map';
+import 'core-js/fn/array/flat-map';
```
I think this can be removed, no?

mturley replied:
Ah, yep. Good catch.

mturley commented:
Big thanks to @michaelkro for the help with cedbe4c ! That helper function is a beast. We'll need to revisit it.
